ast/ast.hpp

```diff
@@ -27,8 +27,11 @@ struct WhileLoop;
 struct ForLoop;
 struct IfStatement;
 struct Expression;
-struct UnaryExpression;
-struct BinaryExpression;
+// struct UnaryExpression;
+// struct BinaryExpression;
```

Comment:
I don't like this merge of Unary/Binary into just Expression. We should actually have a subclass for each kind of operation (`struct MemberAccess: BinaryExpression`, `struct Negation: UnaryExpression`, ...)

Reply:
Is `UnaryExpression`,  `BimaryExpression`, `ModifiablePrimary`, `Primary`  and `RoutineCall` enough for now?

Reply:
Yes they are. I will start planning what other `struct`s do we also need for modification after merging this and the other PR